course_access_groups/acl_backends.py

```diff
@@ -1,19 +1,72 @@
 """
 Access Control backends to implement the Course Access Groups.
 """
-import six
 
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from organizations.api import get_course_organizations
+from organizations.models import UserOrganizationMapping
+from course_access_groups.models import (
+    CourseAccessGroup,
+    GroupCourse,
+    Membership,
+)
 
-def dummy_backend(user, resource, default_has_access, options):  # pylint: disable=unused-argument
+
+def is_organization_staff(user, course):
+    """
+    Helper to check if the user is organization staff.
+
+    TODO: Handle single-site setups in which organization is not important
+    """
+    # TODO: What if a course has two orgs? data leak I guess?
+    organizations = get_course_organizations(course.id)
+    if not organizations:
+        return False
+
+    return UserOrganizationMapping.objects.filter(
+        user=user,
+        organization_id__in=[org['id'] for org in organizations],
+        is_active=True,
+        is_amc_admin=True,
+    ).exists()
+
+
+def is_feature_enabled():
+    """
+    Helper to check Site Configuration for ENABLE_COURSE_ACCESS_GROUPS.
+    :return:
     """
-    Access control backend to prevent users from loading the course `course-v1:Red+Red+Red2020`.
+    return bool(configuration_helpers.get_value('ENABLE_COURSE_ACCESS_GROUPS', default=False))
 
-    Users with emails `@example.com` and `@appsembler.com` are being prevented, as long as they're logged in.
+
+def default_backend(user, resource, default_has_access, options):  # pylint: disable=unused-argument
+    """
+    The Access Control Backend to plug the Course Access Groups feature in Open edX.
     """
-    if six.text_type(resource.id) == 'course-v1:Red+Red+Red2020':
-        if user.is_authenticated and (
-            user.email.endswith('@example.com') or user.email.endswith('@appsembler.com')
-        ):
-            return False
+    if not is_feature_enabled():
+        # Plugin is turned off, maintain the platform's default behaviour.
+        return default_has_access
+
+    if not default_has_access:
+        # Ensures this backend won't leak resources by mistake.
```

Comment:
@OmarIthawi Would you please elaborate on this and do we want to do this check first?

Reply:
I did, in the code.